### PR TITLE
bpo-44661: Update property_descr_set to use vectorcall if possible.

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue44661

````
Mean +- std dev: [property_base] 140 ns +- 5 ns -> [property_vectorcall] 125 ns +- 2 ns: 1.12x faster
````

### CODE REVIEW

**Comment by corona10** [comment]
cc @erlend-aasland

File: Objects/descrobject.c [hunk 1]
```diff
@@ -1633,10 +1633,12 @@ property_descr_set(PyObject *self, PyObject *obj, PyObject *value)
         }
         return -1;
     }
-    if (value == NULL)
+    if (value == NULL) {
         res = PyObject_CallOneArg(func, obj);
-    else
-        res = PyObject_CallFunctionObjArgs(func, obj, value, NULL);
+    } else {
```
Comment (erlend-aasland) [inline Objects/descrobject.c]: Newline before `else` 😉

**Review by erlend-aasland — APPROVED** [review]

**Comment by corona10** [comment]
@erlend-aasland FYI, I updated the wrong benchmark, so I updated the benchmark result :)

**Comment by Fidget-Spinner** [comment]
Out of curiosity, is the benchmark from LTO+PGO build or non-debug build?

**Comment by corona10** [comment]
@Fidget-Spinner 
> Out of curiosity, is the benchmark from LTO+PGO build or non-debug build?

The attached result is tested on a non-debug build but the optimized build also shows a similar result.
````
Mean +- std dev: [property_opt_base] 128 ns +- 5 ns -> [property_opt_vectorcall] 117 ns +- 2 ns: 1.09x faster
````

File: Misc/NEWS.d/next/Core and Builtins/2021-07-17-14-20-59.bpo-44661.BQbXiH.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Update property_descr_set to use vectorcall if possible. Patch by Dong-hee
```
Comment (Fidget-Spinner) [inline Misc/NEWS.d/next/Core and Builtins/2021-07-17-14-20-59.bpo-44661.BQbXiH.rst]:
```suggestion
Update ``property_descr_set`` to use vectorcall if possible. Patch by Dong-hee
```

**Review by Fidget-Spinner — APPROVED** [review]
Wow those are some nice results for a few lines of code changed!

If I understand correctly, this won't slow down any other calls too right? The fast path with vectorcall should always be taken, since in most cases ``func`` should always be a python function/method which supports vectorcall.

**Comment by corona10** [comment]
> this won't slow down any other calls too right?

I believe that this code path is a good example to apply vectorcall :)

**Pull request merged**